Sources/TuistKit/Graph/Graph.swift

```diff
@@ -311,7 +311,7 @@ extension Graph {
     }
 
     // Traverse the graph from the target node using DFS and return all results passing the test.
-    internal func findAll<T: GraphNode>(targetNode: TargetNode, test: (T) -> Bool) -> Set<T> {
+    internal func findAll<T: GraphNode>(targetNode: TargetNode, test: (T) -> Bool, doNotTraverseWhen: (T) -> Bool = { _ in false }) -> Set<T> {
```

Comment:
Nitpick: What about calling the argument `skip`?

Reply:
I called it skip first, and I changed it as I wasn't sure it was obvious enough - maybe I should have trusted my instincts :D